src/librustc/ty/sty.rs

```diff
@@ -1504,6 +1504,46 @@ impl<'a, 'gcx, 'tcx> TyS<'tcx> {
         }
     }
 
+    pub fn conservative_is_uninhabited(&self, tcx: TyCtxt<'a, 'gcx, 'tcx>) -> bool {
+        // Checks whether a type is definitely uninhabited. This is
+        // conservative: for some types that are uninhabited we return `false`,
+        // but we only return `true` for types that are definitely uninhabited.
```

Comment:
Please make this a doc comment (`///`, triple-slashes) above the method.

Also, please mention the relationship with `layout.abi.is_uninhabited()`.